File: javascript/leetcode/map-sum-pairs.js
/**
 * Initialize your data structure here.
 */
var MapSum = function() {
    this.map = {}
};

/** 
 * @param {string} key 
 * @param {number} val
 * @return {void}
 */
MapSum.prototype.insert = function(key, val) {
    this.map[key.toLowerCase()] = val
};

/** 
 * @param {string} prefix
 * @return {number}
 */
MapSum.prototype.sum = function(prefix) {
    res = 0
    p = prefix.toLowerCase()
    for (key in this.map) {
        if (key.indexOf(p) == 0) { res += this.map[key]}
    }
    return res
};

/** 
 * Your MapSum object will be instantiated and called as such:
 * var obj = Object.create(MapSum).createNew()
 * obj.insert(key,val)
 * var param_2 = obj.sum(prefix)
 */

File: java/leetcode/map-sum-pairs.java
class MapSum {

    /** Initialize your data structure here. */
    HashMap<String, Integer> map;

    public MapSum() {
        map = new HashMap<String, Integer>();
    }
    
    public void insert(String key, int val) {
        map.put(key.toLowerCase(), val);
    }
    
    public int sum(String prefix) {
        int res = 0;
        String p = prefix.toLowerCase();
        String thisKey;
        Iterator keys = map.entrySet().iterator();
        while(keys.hasNext()) {
            Map.Entry<String, Integer> pair = (Map.Entry)keys.next();
            if (pair.getKey().indexOf(p) == 0) {
                res += pair.getValue();
            }
        }
        return res;
    }
}

/**
 * Your MapSum object will be instantiated and called as such:
 * MapSum obj = new MapSum();
 * obj.insert(key,val);
 * int param_2 = obj.sum(prefix);
 */
 